react/features/toolbox/components/web/Drawer.js

```diff
@@ -45,29 +45,27 @@ function Drawer({
     onClose,
     t }: Props) {
     const [ expanded, setExpanded ] = useState(false);
-    const drawerRef: Object = useRef(null);
 
     /**
-     * Closes the drawer when clicking or touching outside of it.
+     * Handles clicks within the menu, preventing the propagation of the click event.
      *
-     * @param {Event} event - Mouse down/start touch event object.
+     * @param {Object} event - The click event.
      * @returns {void}
      */
-    function handleOutsideClickOrTouch(event: Event) {
-        if (drawerRef.current && !drawerRef.current.contains(event.target)) {
-            onClose();
-        }
+    function handleInsideClick(event) {
```

Comment:
Why do we need this function to be defined in the component? Isn't it going to be recreated on every rerender? It looks like it doesn't use anything from the component itself.

Question not related to your PR:
I'm wondering why in the first place this component is functional. Since we even have functions defined in it, it looks a lot like class to me. I'm not sure if you were the author of it originally, but do you have any idea?

Reply:
Function components are my preference and I've been slowly adding them here as well. You are correct about the re-rendering, I've wrapped each callback in a `useCallback` hook that takes care of the memoization so there won't be any redefinitions. If you're not a fan of function components I can convert it to a class component, as in this scenario, it's strictly a coding preference of mine.